app/javascript/lib/vue-components/modules/Table.vue

```diff
@@ -0,0 +1,240 @@
+      default: ''
+    },
+    showColumns: {
+      type: Array,
+      default: () => []
+    },
+    routingMember: {
+      type: String,
+      default: ''
+    },
+    routingId: {
+      type: String,
+      default: ''
+    },
+    paginationId: {
+      type: String,
+      default: ''
+    },
+    itemsPerPage: {
+      type: Number,
+      default: 0
+    }
+  },
+  data() {
+    return {
+      map: new Map(),
+      currentSortColumn: this.$options.defaults.sortColumn,
+      currentSort: this.$options.defaults.sortDirection,
+      visibleColumns: this.showColumns,
+      filterColumns: [],
+      dataTable: [],
+      containerPagination: this.paginationId
+    };
+  },
+  computed: {
+    tmpRows() {
+      return this.data || []
+    },
+    rowsSorted() {
+      const id = this.currentSortColumn.field || this.orderColumn;
+      const sort = this.currentSort;
+      return this.tmpRows
+        .slice()
+        .sort(({ [id]: termA }, { [id]: termB }) =>
+          sort === "up"
+            ? typeof termA === "string"
+              ? termA.localeCompare(termB, undefined, { numeric: true })
+              : termA > termB ? -1 : 1
+            : typeof termA === "string"
+              ? termB.localeCompare(termA, undefined, { numeric: true })
+              : termA < termB ? -1 : 1
+        );
+    },
+    icon() {
+      return this.direction === 'down' ? 'down' : 'down-alt'
+    }
+  },
+  created() {
+    this.prepareTable()
+  },
+  methods: {
+    handleTableHeaderClick(id) {
+      const { sort } = this.map.get(id);
+      this.currentSortColumn = id;
+      // toggle sort order
+      this.currentSort = sort === "up" ? "down" : "up";
+      // update the order for the item clicked
+      this.map.set(id, { ...this.map.get(id), sort: this.currentSort });
+    },
+    getSorting(column) {
+      // ignore the first item of the tuple
+      const { sort } = this.map.get(column)
+      return sort;
+    },
+    prepareTable() {
+      this.filterColumns = this.columns.filter(({ field }) => this.visibleColumns.includes(field))
+      this.map.clear();
+      for (let index = 0; index < this.filterColumns.length; index++) {
+        const column = this.filterColumns[index];
+        this.map.set(column, {
+          visibility: this.visibleColumns.includes(column.field),
+          name: column.name,
+          sort: undefined,
+          type: column.type,
+          cssClass: column.cssClass
+        });
+      }
+      this.columnsMapArr = Array.from(this.map);
+    },
+    toggleVisibility({ id, value }) {
+      const columns = this.columns.map(({ field }) => field)
+      const columnName = columns[columns.findIndex((x, index) => index === id)]
+      if (value) {
+        this.visibleColumns.push(columnName)
+      } else {
+        this.visibleColumns = this.visibleColumns.filter(item => item !== columnName)
+      }
+      this.prepareTable()
+      this.$emit('update-show-columns', this.visibleColumns)
+    },
```

Comment:
This cannot be there. `toggleVisibility` is not something that Table.vue manages, but the ColumnSelector